### PR TITLE
Bump to 4.0.0; update curve25519-dalek

### PR DESCRIPTION
**Pull request by conradoplg**
Sorry, I ended up doing a lot of misc stuff in the same PR, but it seems easier to get the release done this way

This also:

- Fixes some issues with the new `pem` feature, and fixes `no_std` support
- Specify MSRV in toolchain file
- Adds CI jobs to to test MSRV and no_std

Closes https://github.com/ZcashFoundation/ed25519-zebra/issues/78

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK

File: tests/util/mod.rs [hunk 1]
```diff
@@ -181,12 +181,10 @@ pub fn order(point: EdwardsPoint) -> &'static str {
...
         }
+    } else if point.is_torsion_free() {
+        "p"
     } else {
-        if point.is_torsion_free() {
-            "p"
-        } else {
-            "8p"
-        }
+        "8p"
```
Comment (daira) [inline tests/util/mod.rs:187]: Pre-existing potential bug: $2p$ and $4p$ are also possible, aren't they?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,12 @@ Entries are listed in reverse chronological order.
 
 # 3.2.0
 
+* Updates `curve25519-dalek` to `4.0.0-rc.2`.
 * Updates `sha2` version to `0.10` and `curve25519-dalek` version to `4.0.0-pre.5`.
```
Comment (daira) [inline CHANGELOG.md]:
```suggestion
* Updates `sha2` version to `0.10`.
```

Comment (daira) [inline CHANGELOG.md]:
```suggestion
* Updates `sha2` version to `0.10`.
```

**Review by daira — APPROVED** [review]
utACK modulo a comment about a pre-existing potential bug (which I don't think needs to be fixed in this release).

File: tests/util/mod.rs (hunk 1 above)
Comment (conradoplg) [reply tests/util/mod.rs:187]: Good catch, I created https://github.com/ZcashFoundation/ed25519-zebra/issues/83 to fix it later

**Review by mpguerra — APPROVED** [review]
approving based on @daira and @nuttycom reviews

**Pull request merged**